### PR TITLE
fix: derive log level from declared field instead of substring match

### LINKED ISSUE #3370: Incorrect log level shown for Argo info log in workflowruns
### Description

The Argo workflow log below is shown with `level: "ERROR"`, even though the actual log message has `level=info` and the log entry itself does not indicate an error.

```json
{
  "timestamp": "2026-05-05T07:56:21Z",
  "log": "time=\"2026-05-05T07:56:21.304Z\" level=info msg=\"sub-process exited\" argo=true error=\"<nil>\"",
  "level": "ERROR"
}
```

### Steps to Reproduce

1. Fetch builds logs of a component by using below curl.
```
curl --location 'http://localhost:8085/api/v1/logs/query' \
--header 'Authorization: Bearer <token>' \
--header 'Content-Type: application/json' \
--data '{
    "startTime": "2026-05-05T07:55:00Z",
    "endTime": "2026-05-07T07:55:00Z",
    "limit": 1000,
    "sortOrder": "asc",
    "searchScope": {
        "namespace": "<ns>",
        "workflowRunName": "<wf_run_name>"
    }
}'
```

2. Check the build logs for the log entry `
"log": "time=\"2026-05-05T07:56:21.304Z\" level=info msg=\"sub-process exited\" argo=true error=\"<nil>\"",`

### Version

v1.0.1-hotfix.1

### Environment Details (with versions)

_No response_

### PR DESCRIPTION
**Pull request by rashadism**
## Problem

Build/workflow logs surface with the wrong level. An Argo line that is `level=info` is reported as `ERROR`:

```json
{
  "timestamp": "2026-05-05T07:56:21Z",
  "log": "time=\"2026-05-05T07:56:21.304Z\" level=info msg=\"sub-process exited\" argo=true error=\"<nil>\"",
  "level": "ERROR"
}
```

The level is re-derived from the raw log text by scanning for level keywords with `strings.Contains`, returning the first match in priority order (ERROR first). Any line whose payload contains a level word — here `error="<nil>"` — is misclassified, regardless of the level the emitter actually declared.

Fixes openchoreo/openchoreo#3370

## Change

Replace the substring scan with a small table of known-shape matchers, tried in order, then a hardened keyword fallback:

- A structural matcher reads the value of a declared `level`/`severity` field (logfmt or JSON), so the declared level is trusted regardless of level words elsewhere on the line.
- A single-letter header matcher (requires the time component) classifies those logs without false-positiving on arbitrary text.
- The keyword fallback matches on token boundaries and ignores a level word used as a field key (`error=`); it runs only on unstructured lines.

## Tests

Table-driven unit tests covering the declared-field formats and value vocabularies, the header shape, the field-key hardening, and edge cases (embedded words, severity precedence, case sensitivity, malformed headers).

## Release Notes

* **Chores**
  * Version bumped to 0.5.2

* **Bug Fixes**
  * Enhanced log level detection to support multiple log formats including structured fields, JSON, logfmt, and klog-style headers
  * Improved parsing accuracy with fallback detection for unstructured logs

### CODE REVIEW

File: observability-logs-opensearch/internal/opensearch/types_test.go [hunk 1]
```diff
@@ -112,15 +112,82 @@ func TestExtractLogLevel(t *testing.T) {
...
+		// Declared logfmt level, trusted over level words in the body.
+		{`time="2026-05-05T07:56:21.304Z" level=info msg="sub-process exited" argo=true error="<nil>"`, "INFO"},
+		{"level=info request failed: connection ERROR", "INFO"},
+		{`level=debug msg="retrying after error"`, "DEBUG"},
+		{`level=error msg="boom"`, "ERROR"},
+		{`level="warn" msg=disk`, "WARN"},
+		{"level = debug starting up", "DEBUG"},
+		{"Level=Error mixed case", "ERROR"},
+		{"lvl=warn cache miss", "WARN"},
+		{"loglevel=debug starting up", "DEBUG"},
```
Comment (akila-i) [inline observability-logs-opensearch/internal/opensearch/types_test.go]: this isnt necessarily a debug log. sounds like a usual startup log printing the value of the environment variable `loglevel`. vice-versa can also happen (eg: if the startup log says `loglevel=error`, its actually a debug or info log, but now flagged as an error. Have we observed `loglevel=<level>` pattern as a way of indicating the actual log level?

Comment (rashadism) [reply observability-logs-opensearch/internal/opensearch/types_test.go]: Dropped loglevel/log_level/log.level. Kept level/lvl/severity/severityText/levelname/@l.

File: observability-logs-opensearch/helm/Chart.yaml [hunk 2]
```diff
@@ -5,8 +5,8 @@ apiVersion: v2
 name: observability-logs-opensearch
 description: A Helm chart for OpenChoreo Observability Logs module with Fluent Bit and OpenSearch
 type: application
-version: 0.5.1
-appVersion: "0.5.1"
+version: 0.5.2
+appVersion: "0.5.2"
```
[the remark this answers is not in the document]
Comment (rashadism) [reply observability-logs-opensearch/helm/Chart.yaml:9]: fixed

File: observability-logs-opensearch/internal/opensearch/types.go [hunk 3]
```diff
@@ -290,20 +291,93 @@ func getStringValue(m map[string]interface{}, key string) string {
...
+	re   *regexp.Regexp
+}{
+	{"ERROR", regexp.MustCompile(`\bERROR\b(?:[^=]|$)`)},
+	{"FATAL", regexp.MustCompile(`\bFATAL\b(?:[^=]|$)`)},
+	{"SEVERE", regexp.MustCompile(`\bSEVERE\b(?:[^=]|$)`)},
+	{"WARN", regexp.MustCompile(`\bWARNING\b(?:[^=]|$)`)},
+	{"WARN", regexp.MustCompile(`\bWARN\b(?:[^=]|$)`)},
+	{"INFO", regexp.MustCompile(`\bINFO\b(?:[^=]|$)`)},
+	{"DEBUG", regexp.MustCompile(`\bDEBUG\b(?:[^=]|$)`)},
+}
```
[the remark this answers is not in the document]
Comment (rashadism) [reply observability-logs-opensearch/internal/opensearch/types.go:327]: fixed

**Review by akila-i — APPROVED** [review]
